core/src/blocks/curl.rs

```diff
@@ -0,0 +1,239 @@
+use crate::blocks::block::{parse_pair, replace_variables_in_string, Block, BlockType, Env};
+use crate::Rule;
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use dns_lookup::lookup_host;
+use hyper::header;
+use hyper::{body::Buf, Body, Client, Method, Request};
+use hyper_tls::HttpsConnector;
+use js_sandbox::Script;
+use lazy_static::lazy_static;
+use pest::iterators::Pair;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+use std::io::prelude::*;
+use url::{Host, Url};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Error {
+    pub error: String,
+}
+
+#[derive(Clone)]
+pub struct Curl {
+    method: String,
+    url: String,
+    headers_code: String,
+    body_code: String,
+}
+
+impl Curl {
+    pub fn parse(block_pair: Pair<Rule>) -> Result<Self> {
+        let mut method: Option<String> = None;
+        let mut url: Option<String> = None;
+        let mut headers_code: Option<String> = None;
+        let mut body_code: Option<String> = None;
+
+        for pair in block_pair.into_inner() {
+            match pair.as_rule() {
+                Rule::pair => {
+                    let (key, value) = parse_pair(pair)?;
+                    match key.as_str() {
+                        "method" => method = Some(value),
+                        "url" => url = Some(value),
+                        "headers_code" => headers_code = Some(value),
+                        "body_code" => body_code = Some(value),
+                        _ => Err(anyhow!("Unexpected `{}` in `curl` block", key))?,
+                    }
+                }
+                Rule::expected => Err(anyhow!("`expected` is not yet supported in `curl` block"))?,
+                _ => unreachable!(),
+            }
+        }
+
+        if !method.is_some() {
+            Err(anyhow!("Missing required `method` in `curl` block"))?;
+        }
+        if !url.is_some() {
+            Err(anyhow!("Missing required `url` in `curl` block"))?;
+        }
+        if !headers_code.is_some() {
+            Err(anyhow!("Missing required `headers_code` in `curl` block"))?;
+        }
+        if !body_code.is_some() {
+            Err(anyhow!("Missing required `body_code` in `curl` block"))?;
+        }
```

Comment:
Same for body, feels like it should be optional, or at least optional for GETs.

Reply:
Same